### ISSUE TITLE
Validate Formfields for nested objects

### ISSUE THREAD

**Issue by theWeird** [sysgears/apollo-universal-starter-kit#1092]
I have a problem to validate formfields of nested objects that are managed in the same form as the main object.

e.g. for management of tasks:
  - Title: Title of Task
  - Desc: Description of Task
  - step[0] Title: Title of first Step
  - step[0] Desc: Description of first Step
  - step[1] Title: Title of second Step
  - step[1] Desc: Description of second Step

It is no problem to send forms like this to the backend. But I cannot find a way to use the current validation method to validate the nested formfields.

**Describe the solution you'd like**
I think it would be great to have a notation like this for the validation of form fields:
- "field": validators for simple field, as it is now
- "nestedList[].field": validators applied for "field" of every item in the nested list
- "nestedList[]": special array validators e.g. for requireing min/max amount of items in that list

**Comment by vlasenko** [comment]
@theWeird Have you studied this code?
https://github.com/sysgears/apollo-universal-starter-kit/blob/master/modules/validation/common-react/validation.ts

**Comment by theWeird** [comment]
Yes, I already tried to figure out if I already can use it for my case.  
Do you think it could already work?  
How would I define the validation entry e.g. for step[i].title?

**Comment by vlasenko** [comment]
@theWeird Why do you think it shouldn't work? I think it should work, based on the code I see there

**Comment by vlasenko** [comment]
@theWeird How about actually trying to read this code and run it in the debugger if you don't understand it. or insert `console.log` ? This code is open, it is small, you can read it and figure out what it is doing
